### PR TITLE
Feature/admin login url

### PR DESCRIPTION
**Pull request by Ethan-Edmond**
# Description

Moves the admin login to only be accessible via url (/login or /admin-dashboard). Changes the admin-dashboard and logout buttons to render conditionally when an admin is logged in and moves them to the navbar. Changes the logout function used by the logout button to use okta's signout method.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Change Status

- [X] Complete, tested, ready to review and merge
- [ ] Complete, but not tested (may need new tests)
- [ ] Incomplete/work-in-progress, PR is for discussion/feedback

# How Has This Been Tested?

- [X] `npm test`

# Checklist

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [ ] My code has been reviewed by at least one peer
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] There are no merge conflicts

### CODE REVIEW

**Comment by Ethan-Edmond** [comment]
[Explanation video](https://www.loom.com/share/15278a7cfeee467f9221b4a15e5b8853)

File: src/components/NavBar/NavBar.js [hunk 1]
```diff
@@ -75,11 +75,18 @@ const NavBar = () => {
           </Link>
         </Menu.Item>
         {authState.isAuthenticated && (
-          <Menu.Item key="5">
-            <div className="menu-link" onClick={logout}>
-              Log out
-            </div>
-          </Menu.Item>
+          <>
```
Comment (ahmedseragcodes) [inline src/components/NavBar/NavBar.js:75]: Hmmm so I think all the changes up until this point make sense, but isn't this moving these buttons / actions to the navbar in the header? If so, I think that's what Jason was saying they don't want, ie they don't want it to be prominent.

Comment (Ethan-Edmond) [reply src/components/NavBar/NavBar.js:75]: it's conditional on this authState.isAuthenticated, so it only shows up if the user is logged in. A non admin shouldn't be able to see the buttons.

Comment (ahmedseragcodes) [reply src/components/NavBar/NavBar.js:75]: Great, got it

**Review by ahmedseragcodes — APPROVED** [review]
Changes follow plan for admin login + reformats code to use okta's native logout that removes token from local storage, no merge conflict, approved.

**Review by imjeremiah — APPROVED** [review]
Changes login to URL, hides login option from non-admin users. Code reformats to use okta. No merge conflicts. Great video explanation, and clarification on new auth conditional. Approved.

**Comment by paulstgermain** [comment]
@Ethan-Edmond I'd love to see the description on this PR filled out appropriately - even with the video walkthrough, it's an important skill to build! As soon as I see this done I will do a full review and look into merging this PR.

File: src/components/Footer/Footer.js [hunk 2]
```diff
@@ -1,25 +1,11 @@
 import React from 'react';
-import { NavLink } from 'react-router-dom';
-import { useOktaAuth } from '@okta/okta-react';
-import { useHistory } from 'react-router-dom';
-import { Button } from 'antd';
```
Comment (paulstgermain) [inline src/components/Footer/Footer.js:5]: Good eye for cleaning up the unused imports in this component upon making this change!

**Review by paulstgermain — APPROVED** [review]
## Video Walkthrough Feedback

Your explanation of the problem you sought to solve was solid, but I would have loved to hear some of the reasoning behind the decision to move the Admin Login feature to be accessible purely via URL. This is an obvious enough case that one could assume it's for security purposes, because who wants to expose the Admin entry point to such a sensitive application to the public, but keep in mind when talking through your work with interviewers in the future that they might not be as quick to connect those points - always be explicit and detailed when describing your work!

As an additional note, and as someone who's looking for you to fully describe the work you did, I also would have *loved* to actually **see** the Route that points to the Admin Login component - but no worries for this specific PR.

Otherwise, you do explain the work you did wonderfully, with enough detail and visual context that I can pick up what you're putting down - so this gets a pass from me!

## Code Review
- Love the eye for detail shown in cleaning up unused imports in the footer upon removing the button that previously allowed us to access the Admin Login component.
- Sensible changes made in not simply copying the content from the footer and into the NavBar. It's clearly written so that only logged-in admins will see these options, and you swapped to a link as opposed to it being a button on _top_ of being a Link.
- The code added here is simple, clean, and is also a nice atomic unit of code accomplishing a specific task - this is also a pass for me!

For the reasons stated above, approving and merging this PR!

**Pull request merged**